src/averageAges.js

```diff
@@ -55,7 +71,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filtered = people.filter(child =>
+    onlyWithSon
+      ? people.find(mother => child.mother === mother.name && child.sex === 'm')
+      : people.find(mother => child.mother === mother.name)
+  );
+
+  const difference = filtered.reduce((sum, child) =>
+    (sum + child.born) - people.find(mother =>
+      mother.name === child.mother).born, 0) / filtered.length;
```

Comment:
It is better to create a variable for found person before to make your code a bit cleaner